File: tests/api/test_contacts_endpoint.py
"""
Test Contacts API Endpoint | Cannlytics API

Author: Keegan Skeate
Contact: <keegan@cannlytics.com>
Created: 7/19/2021
Updated: 7/19/2021
License: MIT License <https://opensource.org/licenses/MIT>
"""
import os
import requests
from dotenv import load_dotenv

# Test using development server.
BASE = 'http://127.0.0.1:8000/api'

# Uncomment to test with production server.
# BASE = 'https://console.cannlytics.com/api'

# Load your API key.
load_dotenv('../../.env')
API_KEY = os.getenv('CANNLYTICS_API_KEY')

# Pass your API key through the authorization header as a bearer token.
HEADERS = {
    'Authorization': 'Bearer %s' % API_KEY,
    'Content-type': 'application/json',
}

# Identify the organization that you are working with.
ORG_ID = 'test-company'

# Define the endpoint.
ENDPOINT = 'contacts'

#------------------------------------------------------------------------------
# Create a contact.
#------------------------------------------------------------------------------
data = {
    'address': '',
    'city': '',
    'contact_id': 'TEST',
    'county': '',
    'email': '',
    'latitude': '',
    'longitude': '',
    'organization': 'Cannlytics Test Contact',
    'phone': '',
    'state': '',
    'street': '',
    'website': '',
    'zip_code': ''
}
url = f'{BASE}/{ENDPOINT}?organization_id={ORG_ID}'
response = requests.post(url, json=data, headers=HEADERS)
assert response.status_code == 200
print('Created:', response.json()['data'])

#------------------------------------------------------------------------------
# Get contacts.
#------------------------------------------------------------------------------
organization_id = 'test-company'
url = f'{BASE}/{ENDPOINT}?organization_id={ORG_ID}'
response = requests.get(url, headers=HEADERS)
assert response.status_code == 200
data = response.json()['data']
print('Found:', len(data))

#------------------------------------------------------------------------------
# Update a contact.
#------------------------------------------------------------------------------
data = {
    'contact_id': 'TEST',
    'city': 'Tulsa',
    'state': 'OK',
}
url = f'{BASE}/{ENDPOINT}?organization_id={ORG_ID}'
response = requests.post(url, json=data, headers=HEADERS)
assert response.status_code == 200
print('Updated:', response.json()['data'])

#------------------------------------------------------------------------------
# Delete a contact.
#------------------------------------------------------------------------------
data = {
    'contact_id': 'TEST',
}
url = f'{BASE}/{ENDPOINT}?organization_id={ORG_ID}'
response = requests.delete(url, json=data, headers=HEADERS)
assert response.status_code == 200
print('Deleted:', response.json()['data'])

File: tests/api/node/test_contacts_endpoint.js
/**
 * Test Contacts Endpoint with Node.js | Cannlytics Console
 * Author: Keegan Skeate
 * Created: 7/25/2021
 * Updated: 7/27/2021
 */
 const axios = require('axios');
 require('dotenv').config();

//  Pass API key through the authorization header
// as a bearer token.
const apiKey = process.env.CANNLYTICS_API_KEY;
const options = {
  headers: { 'Authorization' : `Bearer ${apiKey}` }
};

// Define the API and your organization.
// const base = 'https://console.cannlytics.com/api'; // PRODUCTION:
const base = 'http://127.0.0.1:8000/api'; // DEV: ✓
const orgId = 'test-company';
 
// Create a contact.

let data = {
  address: '',
  city: '',
  contact_id: 'TEST',
  county: '',
  email: '',
  latitude: '',
  longitude: '',
  organization: 'Cannlytics Test Contact',
  phone: '',
  state: '',
  street: '',
  website: '',
  zip_code: ''
};
axios.post(`${base}/contacts?organization_id=${orgId}`, data, options)
  .then(response => {
    console.log(response.data);
  })
  .catch(error => {
    console.log(error);
  });
 
// Get contacts.

axios.get(`${base}/contacts?organization_id=${orgId}`, options)
  .then(response => {
    console.log(response.data);
  })
  .catch(error => {
    console.log(error);
  });
 
// Update a contact.

data = {
  contact_id: 'TEST',
  city: 'Tulsa',
  state: 'OK',
};
axios.post(`${base}/contacts?organization_id=${orgId}`, data, options)
  .then(response => {
    console.log(response.data);
  })
  .catch(error => {
    console.log(error);
  });
 
// Delete a contact.

const objID = 'TEST';
data = { area_id: objID };
axios.delete(`${base}/contacts/${objID}?organization_id=${orgId}`, { data, ...options})
  .then(response => {
    console.log(response.data);
  })
  .catch(error => {
    console.log(error);
  });
 